src/helpers/links.js

```diff
@@ -57,4 +57,26 @@ const isLinkToSelfWithHash = function (href) {
 	return href?.startsWith('#') || href?.startsWith(locationNoHash + '#')
 }
 
-export { domHref, isLinkToSelfWithHash, parseHref }
+/**
+ * Open links, to be used as custom click handler
+ *
+ * @param {string} href the link href
+ */
+const openLink = function (href) {
+	const linkUrl = new URL(href, window.location.href)
+	// Consider rerouting links to Collectives if already inside Collectives app
+	const collectivesUrlBase = '/apps/collectives'
+	if (
+		window.OCA.Collectives?.vueRouter
+		&& linkUrl.pathname.toString().startsWith(generateUrl(collectivesUrlBase))
+	) {
+		const collectivesUrl = linkUrl.href.substring(
+			linkUrl.href.indexOf(collectivesUrlBase) + collectivesUrlBase.length,
+		)
+		window.OCA.Collectives.vueRouter.push(collectivesUrl)
+		return
+	}
```

Comment:
I don't understand what you are doing here. It looks like you are truncating the linkUrl to end with the collectivesUrlBase and then opening it. Do you want to strip the collective name, page and fragments?

Reply:
ahh now i got it. You are using the `substring` **after** the base url. That **does** make sense.